module/os/linux/zfs/zvol_os.c

```diff
@@ -888,14 +888,16 @@ zvol_ioctl(struct block_device *bdev, fmode_t mode,
 		mutex_enter(&zv->zv_state_lock);
 		error = copy_to_user((void *)arg, zv->zv_name, MAXNAMELEN);
 		mutex_exit(&zv->zv_state_lock);
+		if (error)
+			error = SET_ERROR(-error);
 		break;
 
 	default:
-		error = -ENOTTY;
+		error = SET_ERROR(ENOTTY);
```

Comment:
I know that Linux kernel returns negative errors, comparing to FreeBSD.

But it becomes difficult to get error returning logic with this sign manipulations. I am thinking not for this PR, but in future. Will be it better to define something like:
```
#define SET_ERROR_OS ...
```
Which will hide this sign manipulation.
Also this macro will be used only by module/os/* code. Mean, sources compiled only for one operating system. With compile time OS check added explicitly to macro.

Reply:
Yeah, this would be good to do. I've tried this before but it ended up messy, because there's actually a bunch of combinations - where the error came from, where its going, and whether or not we're passing it through or generating. I didn't think super hard though; maybe its not so tricky with a little thought :)